source/grid/grid_tools.cc

```diff
@@ -1573,6 +1573,16 @@ namespace GridTools
       return false;
     }
 
+    template <int dim, typename Cell>
+    inline
+    double calculate_cell_radius(const Cell &cell)
+    {
+      double res = 0.;
+      for (unsigned int v=0; v<GeometryInfo<dim>::vertices_per_cell; ++v)
+        res = std::max(res, ( cell->vertex(v)).distance(cell->center()) );
```

Comment:
@bangerth you are saying that the center may not be located on the diagonal? if so, the original code should be more strict and we should keep it.